Wiki.md

```diff
@@ -0,0 +1,158 @@
+      url "https://plugins.gradle.org/m2/"
+    }
+  }
+}
+
+plugins {
+  // we assume you are already using the Java plugin
+  id "net.ltgt.errorprone" version "0.0.13"
+}
+
+dependencies {
+  annotationProcessor "com.uber.autodispose:autodispose-error-prone-checker:x.y.z" // where x.y.z is the latest version.
+
+  errorprone "com.google.errorprone:error_prone_core:2.1.3"
+}
+
+tasks.withType(JavaCompile) {
+  // Only if you want to support custom configuration
+  // Below is a sample configuration which include Conductor and Activity
+  options.compilerArgs += ["-XepOpt:ClassesWithScope"
+                                       + "=com.bluelinelabs.conductor.Controller,android.app.Activity"]
+}
+```
+
+### Maven
+
+```xml
+<build>
+  <plugins>
+    <plugin>
+      <groupId>org.apache.maven.plugins</groupId>
+      <artifactId>maven-compiler-plugin</artifactId>
+      <version>3.5</version>
+      <configuration>
+        <compilerId>javac-with-errorprone</compilerId>
+        <forceJavacCompilerUse>true</forceJavacCompilerUse>
+        <source>1.8</source>
+        <target>1.8</target>
+        <showWarnings>true</showWarnings>
+        <annotationProcessorPaths>
+          <path>
+             <groupId>com.uber.autodispose</groupId>
+             <artifactId>autodispose-error-prone-checker</artifactId>
+             <version>x.y.z</version>
+          </path>
+        </annotationProcessorPaths>
+        <compilerArgs>
+          <!-- Only if you want to support custom configuration
+          Below is a sample configuration which include Conductor and Activity -->
+          <arg>--XepOpt:ClassesWithScope=com.bluelinelabs.conductor.Controller,android.app.Activity</arg>
+        </compilerArgs>
+      </configuration>
+      <dependencies>
+        <dependency>
+          <groupId>org.codehaus.plexus</groupId>
+          <artifactId>plexus-compiler-javac-errorprone</artifactId>
+          <version>2.8</version>
+        </dependency>
+        <!-- override plexus-compiler-javac-errorprone's dependency on
+             Error Prone with the latest version -->
+        <dependency>
+          <groupId>com.google.errorprone</groupId>
+          <artifactId>error_prone_core</artifactId>
+          <version>2.1.3</version>
+        </dependency>
+      </dependencies>        
+    </plugin>
+</build>
+```
+
+## Code example
+
+```java
+public class ComponentWithLifecycle extends Activity {
+  public void observeOnSomething() {
+    Observable
+        .interval(1, TimeUnit.SECONDS)
+        .subscribe(new Consumer<Long>() {
+          @Override public void accept(Long interval) throws Exception {
+            System.out.println(interval);
+          }
+        });
+  }
+}
+```
+
+```
+./gradlew build
+error: [UseAutoDispose] Always apply an AutoDispose scope before subscribing within defined scoped elements.
+        .subscribe(new Consumer<Long>() {
+                  ^
+    (see https://github.com/uber/AutoDispose/wiki/Error-Prone-Checker)
+```
```

Comment:
Space after code blocks too. This should be human readable before and after rendering

Reply:
https://github.com/uber/AutoDispose/pull/162/commits/7412a41a4e6dcd8ff1214b95bea62458756cfe57